src/main/java/sparksoniq/spark/iterator/function/StructuredJsonFileFunctionIterator.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Authors: Stefan Irimescu, Can Berker Cikis
+ *
+ */
+
+package sparksoniq.spark.iterator.function;
+
+import org.apache.spark.SparkException;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import sparksoniq.exceptions.SparksoniqRuntimeException;
+import sparksoniq.jsoniq.runtime.iterator.RDDRuntimeIterator;
+import sparksoniq.jsoniq.runtime.iterator.RuntimeIterator;
+import sparksoniq.jsoniq.runtime.metadata.IteratorMetadata;
+import sparksoniq.semantics.DynamicContext;
+import sparksoniq.spark.SparkSessionManager;
+
+import java.util.List;
+
+public class StructuredJsonFileFunctionIterator extends RDDRuntimeIterator {
+
+    private static final long serialVersionUID = 1L;
+
+    public StructuredJsonFileFunctionIterator(List<RuntimeIterator> arguments, IteratorMetadata iteratorMetadata) {
+        super(arguments, iteratorMetadata);
+    }
+
+    @Override
+    public void open(DynamicContext context) {
+        super.open(context);
+
+        long resultSize = this.getRDD(_currentDynamicContext).count();
+        this._hasNext = resultSize != 0;
```

Comment:
Why is this needed?

Reply:
Great catch! After taking a look at this I have noticed that this redundant override exists for not only structuredJSON file but all of RDDRuntimeIterator implementations. I have removed them as well and updated the PR.

Reply:
Very nice job! Thanks.